server/controllers/UserController.js

```diff
@@ -0,0 +1,38 @@
+import models from '../database/models';
+import helpers from '../helpers';
+
+const { Book, LendingHistory, Author } = models;
+const { responseMessage } = helpers;
+
+/**
+ * @class UserController
+ */
+export default class UserController {
+  /**
+   * @description allows users to borrow a book
+   * @param {object} request express request object
+   * @param {object} response express response object
+   * @returns {json} json
+   * @memberof UserController
+   */
+  static async borrowBook(request, response) {
+    try {
+      const { id } = request.userData;
+      const { duration, bookId } = request.body;
+      const existingBook = await Book.findOne({ where: { id: bookId }, include: [Author] });
+      if (!existingBook) return responseMessage(response, 404, { message: 'book not found' });
+      const { dataValues } = existingBook;
+      const bookStock = dataValues.stock;
+      const { fullname } = dataValues.Author;
+      delete dataValues.authorId;
+      delete dataValues.Author;
+      if (!bookStock) return responseMessage(response, 200, { status: 'success', message: 'book is out stock', data: { ...dataValues, authorName: fullname } });
```

Comment:
this will never do anything.

Reply:
Kindly test this route, I have tested it and it does something or why do you think it would not do anything